app/src/main/java/org/mozilla/fenix/settings/search/SearchStringValidator.kt

```diff
@@ -4,57 +4,33 @@
 
 package org.mozilla.fenix.settings.search
 
+import android.content.Context
 import android.net.Uri
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.Response
 import mozilla.components.support.ktx.kotlin.toNormalizedUrl
-import org.mozilla.fenix.ext.logDebug
-import java.io.IOException
-import java.net.HttpURLConnection
-import java.net.MalformedURLException
-import java.net.URL
+import org.mozilla.fenix.ext.components
 
 object SearchStringValidator {
     enum class Result { Success, MalformedURL, CannotReach }
 
-    fun isSearchStringValid(searchString: String): Result {
-        val searchURL = createSearchURL(searchString) ?: return Result.MalformedURL
-        val connection = openConnection(searchURL)
-
-        return try {
-            if (connection.hasValidResponseCode) Result.Success else Result.CannotReach
-        } catch (e: IOException) {
-            logDebug(LOGTAG, "Failure to get response code from server: returning invalid search query")
-            Result.CannotReach
-        } finally {
-            try { connection.inputStream.close() } catch (_: IOException) {
-                logDebug(LOGTAG, "connection.inputStream failed to close")
-            }
-
-            connection.disconnect()
-        }
+    fun isSearchStringValid(context: Context, searchString: String): Result {
+        val request = createRequest(searchString) ?: return Result.MalformedURL
+        val response = context.components.core.client.fetch(request)
+        return if (response.hasValidResponseCode) Result.Success else Result.CannotReach
```

Comment:
nit: `response.isSuccess` does the same as `hasValidResponseCode`

Reply:
Good to know!